apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -939,7 +951,20 @@ process_outgoing_stanza(El, StateData) ->
                 _ ->
                     NewEl1
             end,
-    NState = process_outgoing_stanza(ToJID, Name, {Attrs, NewEl, FromJID, StateData, Server, User}),
+    Acc2 = mongoose_acc:put(element, NewEl, Acc1),
+    Name = mongoose_acc:get(name, Acc2),
+    NState = case Name of
+        <<"presence">> ->
+            % new-style
+            % I think it still makes sense to pass all those arguments, for performance reasons
```

Comment:
Also since, you already changed the implementation for `presence` stanza only, maybe it would be good to rename the function to sth like `process_outgoing_presence`?